go/vt/mysqlctl/backupengine.go

```diff
@@ -381,40 +381,65 @@ func addMySQL8DataDictionary(fes []FileEntry, base string, baseDir string) ([]Fi
 func findFilesToBackup(cnf *Mycnf) ([]FileEntry, int64, error) {
 	var err error
 	var result []FileEntry
-	var totalSize int64
-
-	// first add inno db files
-	result, totalSize, err = addDirectory(result, backupInnodbDataHomeDir, cnf.InnodbDataHomeDir, "")
-	if err != nil {
-		return nil, 0, err
-	}
-	result, size, err := addDirectory(result, backupInnodbLogGroupHomeDir, cnf.InnodbLogGroupHomeDir, "")
-	if err != nil {
-		return nil, 0, err
+	var size, totalSize int64
+	var flavor MySQLFlavor
+	var version ServerVersion
+	var features capabilitySet
+
+	// get the flavor and version to deal with any behavioral differences
+	{
```

Comment:
nit: we have tried to avoid this style of using `{ ... }` to delineate blocks of code. The comments in between the various blocks should be sufficient.